### ISSUE TITLE
Assignment to Final variable shouldn't be allowed using an augmented assignment

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3099]
This should generate an error, but it doesn't:

```python
from typing import Final
test_map: Final = {"4": "5"}
test_map |= {"a": "b"}
```

This was initially reported as a part of another bug report: https://github.com/microsoft/pyright/issues/3098

**Comment by erictraut** [comment]
This will be addressed in the next release.
